sbi/inference/posteriors/mcmc_posterior.py

```diff
@@ -0,0 +1,548 @@
+        num_chains = _maybe_use_dict_entry(num_chains, "num_chains", m_p)
+        init_strategy = _maybe_use_dict_entry(init_strategy, "init_strategy", m_p)
+        init_strategy_num_candidates = _maybe_use_dict_entry(
+            init_strategy_num_candidates, "init_strategy_num_candidates", m_p
+        )
+        self.potential_ = self._prepare_potential(method)  # type: ignore
+
+        init_fn = self._build_mcmc_init_fn(
+            self.proposal, self.potential_fn, transform=self.theta_transform
+        )
+        initial_params = torch.cat(
+            [init_fn() for _ in range(num_chains)]  # type: ignore
+        )
+
+        num_samples = torch.Size(sample_shape).numel()
+
+        track_gradients = method in ("hmc", "nuts")
+        with torch.set_grad_enabled(track_gradients):
+            if method in ("slice_np", "slice_np_vectorized"):
+                transformed_samples = self._slice_np_mcmc(
+                    num_samples=num_samples,
+                    potential_function=self.potential_,
+                    initial_params=initial_params,
+                    thin=thin,  # type: ignore
+                    warmup_steps=warmup_steps,  # type: ignore
+                    vectorized=(method == "slice_np_vectorized"),
+                    show_progress_bars=show_progress_bars,
+                )
+            elif method in ("hmc", "nuts", "slice"):
+                transformed_samples = self._pyro_mcmc(
+                    num_samples=num_samples,
+                    potential_function=self.potential_,
+                    initial_params=initial_params,
+                    mcmc_method=method,  # type: ignore
+                    thin=thin,  # type: ignore
+                    warmup_steps=warmup_steps,  # type: ignore
+                    num_chains=num_chains,
+                    show_progress_bars=show_progress_bars,
+                ).detach()
+            else:
+                raise NameError
+
+        samples = self.theta_transform.inv(transformed_samples)
+        return samples.reshape((*sample_shape, -1))
+
+    def _build_mcmc_init_fn(
+        self,
+        proposal: Any,
+        potential_fn: Callable,
+        transform: torch_tf.Transform,
+        init_strategy: str = "prior",
+        **kwargs,
+    ) -> Callable:
+        """Return function that, when called, creates an initial parameter set for MCMC.
+
+        Args:
+            proposal: Proposal distribution.
+            potential_fn: Potential function that the candidate samples are weighted
+                with.
+            init_strategy: Specifies the initialization method. Either of
+                [`prior`|`sir`|`latest_sample`].
+            kwargs: Passed on to init function. This way, init specific keywords can
+                be set through `mcmc_parameters`. Unused arguments should be absorbed.
+
+        Returns: Initialization function.
+        """
+        if init_strategy == "prior":
```

Comment:
now that we introduced the notion of `proposal` for the mcm init, it might be confusing that we do the prior init with the proposal. how do we know it's actually the prior? 
shouldn't we interpret it as different strategies for sampling from the proposal? e.g., "just sampling from it" (now "prior"), vs. performing sir (currently "sir")? 
maybe sth for a future PR / issue for now.

Reply:
in other words, the init strategy could be defined as the way of sampling from the proposal. whether it's the prior or something else should be defined solely by `proposal`, no?

Reply:
Yes, I agree. How about renaming `"prior"` to `"direct"`?